result_iterator.go

```diff
@@ -48,15 +47,11 @@ func (r *queryResultIterator) More() bool {
 	}
 
 	if r.results == nil {
-		select {
-		case <-r.ctx.Done():
+		results, ok := <-r.query.Ready()
+		if !ok {
```

Comment:
It doesn't seem like `r.done` gets set to `true` until `Release` is actually called.  This might be surprising to future-us?  Maybe `r.done` should be called `r.released`?

Reply:
Done.